### PR TITLE
S3 storage i/support remote sources

### PR DESCRIPTION
**Pull request by shimonmagal**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Run me with docker run -v <host-path-to-storage>:/opt/takipi-storage/storage -p <host_port>:$STORAGE_PORT
-# Logs are written to <storage>/log folder and both should be persistent outside of the running container
```
Comment (chook) [inline docker/Dockerfile:2]: - [x] the both here is (a) logs folder, (b) storage folder from previous line

File: README.md [hunk 2]
```diff
@@ -1,10 +1,13 @@
 Takipi Storage
 ==============
 
+With thanks to moovel for developting this version supporting s3: https://github.com/moovel/takipi-storage/tree/s3-storage
```
Comment (chook) [inline README.md]: - [x] typo in developting + grammer

File: docker/README.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 takipi-storage on docker
 ========================
 
-When running takipi-storage inside docker, make sure to persist the storage folder outside (using volumes).
+To run takipi-storage for s3:
+- configure bucket, access key and password in settings.yml
```
Comment (chook) [inline docker/README.md]: - [x] In AWS jargon this is secret and not password

File: docker/settings.yml [hunk 4]
```diff
@@ -1,27 +1,36 @@
...
+    bucket: <bucket>
+    credentials:
+      accessKey: <key> 
+      secretKey: <secret> 
+
 server:
   applicationConnectors:
     - type: http
       port: 8080
+# this requires the alpn-boot library on the JVM's boot classpath
```
Comment (chook) [inline docker/settings.yml]: - [x] it's time we remove this spdy nonesense and change it to a https 8443 commented out

File: src/main/java/com/takipi/oss/storage/fs/SimplePathRecord.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
+package com.takipi.oss.storage.fs;
+
+public class SimplePathRecord  implements BaseRecord {
```
Comment (chook) [inline src/main/java/com/takipi/oss/storage/fs/SimplePathRecord.java]: - [x] double space before implements

File: src/main/java/com/takipi/oss/storage/fs/SimplePathRecord.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Override
+    public String getPath() {
+        return path;
+    }
+
+    @Override
+    public String getServiceId() {
+        if (pathParts.length > 0)
+        {
```
Comment (chook) [inline src/main/java/com/takipi/oss/storage/fs/SimplePathRecord.java]: - [x] Our OSS is the regular Java one, which means the bracket should be inlined

File: src/main/java/com/takipi/oss/storage/fs/Record.java [hunk 7]
```diff
@@ -33,6 +33,10 @@ public String getKey() {
         return key;
     }
 
+    public String getPath() {
+       return  this.getServiceId() + "/" + this.getType() + "/" + this.getKey();
```
Comment (chook) [inline src/main/java/com/takipi/oss/storage/fs/Record.java]: - [x] perhaps we should move "/" to a constant (or use File.separator)

**Review by chook — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (shimonmagal) [reply docker/Dockerfile:2]: Sure, but there is no actual storage folder in this case

File: settings.yml [hunk 8]
```diff
@@ -1,17 +1,28 @@
-folderPath: /opt/takipi-storage/storage
-maxUsedStoragePercentage: 0.95
 enableCors: true
 corsOrigins: "*"
 
+# If using attaching IAM Role to instance leave accessKey and secretKey empty.
```
Comment (chook) [inline settings.yml]: - [x] why are settings.yml defaulting to s3?

File: src/main/java/com/takipi/oss/storage/TakipiStorageMain.java [hunk 9]
```diff
@@ -96,14 +95,28 @@ private Filesystem configureS3Filesystem(TakipiStorageConfiguration configuratio
 
         // Setup Amazon S3 client
         TakipiStorageConfiguration.S3Fs.Credentials credentials = configuration.getS3Fs().getCredentials();
-        AWSCredentials awsCredentials = new BasicAWSCredentials(credentials.getAccessKey(), credentials.getSecretKey());
-        AmazonS3 amazonS3 = new AmazonS3Client(awsCredentials);
+
+        AmazonS3 amazonS3;
+        
+        if ( credentials.getAccessKey() != null && !credentials.getAccessKey().isEmpty() &&
```
Comment (chook) [inline src/main/java/com/takipi/oss/storage/TakipiStorageMain.java]: styles

File: settings.yml (hunk 8 above)
Comment (chook) [reply settings.yml]: - [x] agreed that in the base branch of this PR, the default is still S3.

**Comment by chook** [comment]
my eclipse is going crazy with warnings about Filesystem the raw type. @shimonmagal what's up with that?

**Pull request merged**